src/Component/src/Metadata/Resource/Factory/OperationDefaultsTrait.php

```diff
@@ -0,0 +1,142 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Sylius Sp. z o.o.
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace Sylius\Resource\Metadata\Resource\Factory;
+
+use Sylius\Resource\Metadata\HttpOperation;
+use Sylius\Resource\Metadata\MetadataInterface;
+use Sylius\Resource\Metadata\Operation;
+use Sylius\Resource\Metadata\RegistryInterface;
+use Sylius\Resource\Metadata\ResourceMetadata;
+use Sylius\Resource\Symfony\Request\State\Responder;
+use Sylius\Resource\Symfony\Routing\Factory\RouteName\OperationRouteNameFactory;
+
+/**
+ * @internal
+ */
+trait OperationDefaultsTrait
+{
+    private function getResourceWithDefaults(
+        string $resourceClass,
+        ResourceMetadata $resource,
+        MetadataInterface $resourceConfiguration,
+    ): ResourceMetadata {
+        $resource = $resource->withClass($resourceClass);
+
+        if (null === $resource->getAlias()) {
+            $resource = $resource->withAlias($resourceConfiguration->getAlias());
+        }
+
+        if (null === $resource->getApplicationName()) {
+            $resource = $resource->withApplicationName($resourceConfiguration->getApplicationName());
+        }
+
+        if (null === $resource->getName()) {
+            $resource = $resource->withName($resourceConfiguration->getName());
+        }
+
+        return $resource;
+    }
+
+    private function getOperationWithDefaults(
+        OperationRouteNameFactory $operationRouteNameFactory,
+        RegistryInterface $resourceRegistry,
+        ResourceMetadata $resource,
+        Operation $operation,
+    ): array {
```

Comment:
Does it makes sense to re-sort method arguments by importance?

```suggestion
    private function getOperationWithDefaults(
        Operation $operation,
        ResourceMetadata $resource,
        OperationRouteNameFactory $operationRouteNameFactory,
        RegistryInterface $resourceRegistry,
    ): array {
```